### ISSUE TITLE
Out of date with the newest client 

### ISSUE THREAD

**Issue by geoeo** [cralor/TourGuideVanilla#43]
tldr:

<img width="557" alt="out_of_date" src="https://user-images.githubusercontent.com/1128542/63459562-0bc16a00-c455-11e9-87da-5bc7e0fea88e.png">

**Comment by cralor** [comment]
Yup... this addon isn't updated to the Classic API.

I'll get to it, but not before launch. It will most likely depend on how much I feel like I want it when I start leveling for real :)

**Comment by geoeo** [comment]
ok cool. just wanted to let you know. What exactly does that mean though?

**Comment by cralor** [comment]
This version of addon was designed for the "original" Vanilla WoW 1.12.1 and not 1.13... aka Classic WoW. Therefore the code is out-of-date and does not work.

**Comment by geoeo** [comment]
so its unusable for 1.13?

**Comment by cralor** [comment]
Correct

**Comment by Sefriol** [comment]
You could probably use the original Tour Guide, since it was made for TBC i.e. much newer API. It just lacks some calculations since they added 10% XP buff for TBC. I might test that out when login queues get lower.

**Comment by saghloqh** [comment]
Hi, now that the games has dropped, I was wondering if there are any thoughts. None of the AddOns out there available for 1.13 really solve the problem of guide creation very well. I remember using TourGuide on a private server and liking it. Finding out it's not compatible with classic was a shock!

I have been wanting to create a guide AddOn and website for a while - not only for leveling, but for a whole bunch of different things. Basically users could create guides for just about everything (leveling, gathering routes, professions, gold grinding, literally whatever) in a fairly accessible way, then everything would be shared through a website.

The quest guides are the most interesting part though, and Tour Guide does it extremely well. If it got back up and running, my project would be put aside for a long time because I don't really need to do it... the extras are just an excuse to get the quest guides working.

Anyway, I'm going to try to fix up the errors and make the AddOn actually useable for myself. I won't share it unless I get cralor's blessing, but I also just wanted to throw forward that I'd be willing to help update TourGuide to 1.13 if you're interested Cralor.

**Comment by Sefriol** [comment]
TekNoLogic's TourGuide seems to be a good starting point. I tested it on Classic and it does not raise any major errors (even shows up the greeting message), but it does not show up normally when running the game.

I did not do full diagnostics on why this is since I don't have my typical bug grabbers enabled, but it might be because of the quest data and game data do not match. If this is the case, it should not be a major issue.

There might be some APIs that have been disabled since addon was last updated (Cataclysm), which probably need remapping.

**Comment by saghloqh** [comment]
That seems like a positively amazing starting point. I'm going to start doing some poking around with it and see what's what (starting in a couple hours).

For the time being what I'm doing is just so my friends and I can do some powerleveling and play at least some of the endgame. We aren't the teenagers we were once!

TekNoLogic hasn't been active for like a decade or something, so I can't imagine him being mad about me fixing bugs and version incompatibilities, However, I will still message him just in case and as I said before, eventually I going be creating my own AddOn, and I'd just redo everything at that point.

**Comment by Sefriol** [comment]
His Github feed seems to indicate that he has some activity even with WoW. I doubt he would be mad if you create a fork and submit a pull request.

I also tried to check if the Burning Crusade version works, but I think it has more problems than the Cata version. Probably the best match would be to integrate TBC guides as a starting point and use Cata version otherwise.

**Comment by cralor** [comment]
Yeah, I'm honestly going to fore-go leveling to get this guy up and running :)

I've looked at some classic updated addons and still prefer TourGuide :-)

Let me also pull up the Cata version and start poking around!

**Comment by cralor** [comment]
Starting it as a new project here: https://github.com/cralor/TourGuideClassic

**Comment by Sefriol** [comment]
Made an initial quick and dirty fix to make TourGuide UI atleast visible:
https://github.com/Sefriol/TourGuide/tree/classic

**Comment by cralor** [comment]
Cool I’ll take a look

Did you see my new repo for it?

**Comment by cralor** [comment]
I just took a look at your edits. Looks good. I made most of the same ones. I'll compare more closely to see if you introduced any improvements.

I got it to a good state. The only issue I was having was it was automatically marking Quest Accept tasks as complete without actually having picked them up.

**Comment by Sefriol** [comment]
I would just base the repo on the TekNoLogic's repo so we can have the commit history and can easily merge it on top of it incase he wants to continue development.

**Comment by cralor** [comment]
Good idea

**Comment by Sefriol** [comment]
> I got it to a good state. The only issue I was having was it was automatically marking Quest Accept tasks as complete without actually having picked them up.

Pretty much ended in the same spot. For the majority of the tasks it marks them automatically completed. Did not manage to debug the reason for now, but there are couple of things I noticed:

1. If you can untag the quest, it will also untag the quest pickup
2. For some quest you cannot even untag them even if the objective is not complete

This would lead to me believe that there is some event loop that marks this event complete all the time.

**Comment by cralor** [comment]
Yup. Can you help me look into getting his debugging setup running?

It has a dependency for his tekDebug addon but it has its own issues. That
would help us haha.

**Comment by Sefriol** [comment]
Debugger seems to work. Just use /td instead of the button within UI. I have very limited time, but I try to see if I can figure this out now.

**Comment by Sefriol** [comment]
Okey, I narrowed it down. Quest Objectives for those quests are not found, so they are marked as complete.

**Comment by Sefriol** [comment]
It was little bit different than I thought, but quite close.

https://github.com/Sefriol/TourGuide/commit/a5b5fd03bdb087e18444246171d6f2d9fdf6f49f

Does not fix all guide autocompletes, but does well for the major part. Needs more testing though.

https://wow.gamepedia.com/API_GetQuestLogTitle was changed in WoD, which caused the quest isComplete check to be in incorrect position. It should have been 6 where as 7 is now populated by "frequency".

**Comment by cralor** [comment]
I did notice that questlogtitle issue as well.

Sorry I didn’t setup the Git well, would be better to show you my diff. My
apologies.

Thanks though I’ll take a look!

**Comment by Sefriol** [comment]
You could forcefully fix the git by using following method:

1. Fork/Clone the original version https://github.com/TekNoLogic/TourGuide
2. Copy paste your https://github.com/cralor/TourGuideClassic files on top of it
3. Commit changes (preferably so that each logical change is documented via multiple commits. You can use `git add --patch` for that)
4. Edit git config origin to point to https://github.com/cralor/TourGuideClassic (`git config --edit`)
5. force push it (git push --force`)

This should restore the commit history and make it easy to see what changes you did. (You might need to map your local master to origin/master. Not sure since I'm writing this out of memory) You can also play it save by doing this to brand new repository. Then when it plays out correctly, you can delete TouGuideClassic repository and rename the new one into it.

**Comment by cralor** [comment]
Awesome thanks for that. I went ahead and made those changes.
